evosax/strategies/lm_ma_es.py

```diff
@@ -158,7 +138,7 @@ def ask_strategy(
             state.c_d,
             state.gen_counter,
         )
-        return x, state
+        return x, state.replace(z=z)
```

Comment:
Maybe I am missing it, but where is the `z` needed? `(popsize, num_dims)` will require quite a lot of memory for many applications. Can we reconstruct this from the sampled `x` candidates instead?

Reply:
Yes, this modification is the largest difference between original evosax's code and numpy code for memory size.
We have to use ```z``` to update ```p_sigma,M```(pls see the update functions), but in order to get ```z``` from solution vectors we have to calculate inverse matrices and that may causes both computing errors and cost. I think the procedure for getting z from solutions will be like this;
  iteratively calculate inverse of``` (1-c_d[j])*I+c_d[j]*jnp.outer(M[j,:],M[j,:])``` like ```sample``` function and select j by checking   generation.


Are there any linear algebra tricks to keep away from calculation of inverse matrices? I can't came up with that.

Reply:
Mhmm, I am not sure -- but I guess this is a bit of a problem. I guess it will be hard to scale to neuroevolution even though it is the limited memory version :( I will give this more of a thought.

Reply:
Hello. If possible, could you please approve the workflow and run the test before devising the implementation? I am wondering if there are any implementation mistakes at this time.